### PR TITLE
Adding troubleshooting sections for NFS

### PR DESCRIPTION
**Pull request by asettle**

### CODE REVIEW

File: xml/bp_troubleshooting_nfs.xml [hunk 1]
```diff
@@ -13,7 +13,10 @@
    <dm:release>SES 7</dm:release>
   </dm:docmanager>
  </info>
- <para>
-  TBD
- </para>
+ <sect1 xml:id="bp-troubleshooting-debug-logging">
+   <title>Debugging Logging</title>
```
Comment (asettle) [inline xml/bp_troubleshooting_nfs.xml:17]: $ cat userconf-nfs.foo                ​
NFS_CORE_PARAM { ​
       Enable_NLM = false; ​
       Enable_RQUOTA = false; ​
       Protocols = 4; ​
       NFS_Port = 12345; ​
}​
 ​

$ cat conf-nfs.foo                ​
%url "rados://nfs-ganesha/foo/export-1" ​
%url "rados://nfs-ganesha/foo/userconf-nfs.foo" ​

​

$ rados --pool nfs-ganesha --namespace foo put userconf-nfs.foo userconf-nfs.foo​
$ rados --pool nfs-ganesha --namespace foo put conf-nfs.foo conf-nfs.foo​

​

$ ceph orch restart nfs.foo ​
restart nfs.foo.node1 from host 'node1' ​
restart nfs.foo.node3 from host 'node3'

File: xml/bp_troubleshooting_nfs.xml [hunk 2]
```diff
@@ -13,7 +13,10 @@
...
+ </sect1>
+ <sect1 xml:id="bp-troubleshooting-default-port">
+   <title>Change Default Port</title>
```
Comment (asettle) [inline xml/bp_troubleshooting_nfs.xml:20]: $ cat userconf-nfs.foo                ​
NFS_CORE_PARAM { ​
       Enable_NLM = false; ​
       Enable_RQUOTA = false; ​
       Protocols = 4; ​
       NFS_Port = 12345; ​
}​
 ​

$ cat conf-nfs.foo                ​
%url "rados://nfs-ganesha/foo/export-1" ​
%url "rados://nfs-ganesha/foo/userconf-nfs.foo" ​

​

$ rados --pool nfs-ganesha --namespace foo put userconf-nfs.foo userconf-nfs.foo​
$ rados --pool nfs-ganesha --namespace foo put conf-nfs.foo conf-nfs.foo​

​

$ ceph orch restart nfs.foo ​
restart nfs.foo.node1 from host 'node1' ​
restart nfs.foo.node3 from host 'node3'


---------------------------------


ss -anlp | grep ganesha               ​
udp        UNCONN       0       0         *:12345       *:*       users:(("ganesha.nfsd",pid=21271,fd=46)) ​
tcp        LISTEN       0       128       *:12345       *:*       users:(("ganesha.nfsd",pid=21271,fd=47))​

​

mount -t nfs -o port=12345 node1:/bar /mnt​

​

$ mount -t nfs node1:/bar /mnt​

$ ls -lah /mnt/​

total 7.5K​

drwxr-xr-x  2 root root 2.9K Sep 22 00:47 .​

drwxr-xr-x 23 root root 4.0K Aug 29 02:26 ..​

-rw-r--r--  1 root root  994 Sep 22 00:47 MAGIC.TXT​

-rw-r--r--  1 root root  900 Sep 22 00:47 PUFF.TXT​

-rw-r--r--  1 root root  974 Sep 22 00:47 SEA.TXT

File: xml/bp_troubleshooting_nfs.xml [hunk 3]
```diff
@@ -13,7 +13,35 @@
...
   </dm:docmanager>
  </info>
- <para>
-  TBD
- </para>
+ <sect1 xml:id="bp-troubleshooting-debug-logging">
+   <title>Logging</title>
+   <para>
+     As of &productname; &productnumber;, &ganesha; reads the configuration
+     data from <filename>/etc/ganesha/ganesha.conf</filename>.
```
Comment (mgfritch) [inline xml/bp_troubleshooting_nfs.xml:20]:
```suggestion
     data from <filename>/etc/sysconfig/ganesha</filename>.
```

In SES6 this setting was actually from `/etc/sysconfig/ganesha`, but that no longer exists in SES7.

We need to find a replacement that is either:
- write a LOG block into a RADOS user config
- add log level to the NFS ServiceSpec?

File: xml/bp_troubleshooting_nfs.xml [hunk 4]
```diff
@@ -13,7 +13,35 @@
...
+   </para>
+   <sect2 xml:id="troubleshooting-set-log-level">
+     <title>Setting the Log Level</title>
+     <para>
+       You change the default debug level by ....., and replacing with ...
+     </para>
+     <para>
+       To debug logging, change the log level to <literal>FULL_DEBUG</literal>
+       to increase the log verbosity. Keep in mind that this can produce
+       large amounts of data in the log files.
```
Comment (mgfritch) [inline xml/bp_troubleshooting_nfs.xml:37]:
```suggestion
       large amounts of data in journalctl and adversely affect performance.
```

File: xml/bp_troubleshooting_nfs.xml (hunk 3 above)
Comment (asettle) [reply xml/bp_troubleshooting_nfs.xml:20]: Why have you reverted it back to the SES 6 location?

Comment (mgfritch) [reply xml/bp_troubleshooting_nfs.xml:20]: we might want to redo this entire paragraph. `/etc/sysconfig/ganesha` does not exist anymore and `/etc/ganesha/ganesha.conf` live in the container data dir and shouldn't really be modified (bootstrap code only).

The log level is current set in the `unit.run` file, but I think we should discourage modifying that also.

I'll work towards a PR that adds a log level (e.g. daemon params) to the service spec which can act as a replacement for the sysconfig file ...

File: xml/bp_troubleshooting_nfs.xml [hunk 5]
```diff
@@ -13,7 +13,131 @@
...
+     </para>
+<screen>
+  LOG { ​
+  # The components block contains one or more logging components ​
+  # and the setting to be used. ​
+  components { ​
+  # The ALL component is special. When set it defines the level ​
+  # for all components and overrides any other setting in this block. ​
+  ALL = FULL_DEBUG; # this will likely kill performance ​
+  } ​
```
[the remark this answers is not in the document]
Comment (mgfritch) [reply xml/bp_troubleshooting_nfs.xml:67]: missing param and add some space?

File: xml/bp_troubleshooting_nfs.xml [hunk 6]
```diff
@@ -13,7 +13,131 @@
...
+</screen>
+     <para>
+       Put the file into &rados;. In this example, the file is named
```
Comment (mgfritch) [inline xml/bp_troubleshooting_nfs.xml:70]:
```suggestion
       Put the file into a &rados; configuration object. In this example, the file is named
```

File: xml/bp_troubleshooting_nfs.xml [hunk 7]
```diff
@@ -13,7 +13,131 @@
...
+       <filename>debugconf-nfs.foo</filename>:
+     </para>
+<screen>
+%url "rados://nfs-ganesha/foo/export-1" ​
+%url "rados://nfs-ganesha/foo/userconf-nfs-foo" ​
+%url "rados://nfs-ganesha/foo/debugconf-nfs.foo"​
```
Comment (mgfritch) [inline xml/bp_troubleshooting_nfs.xml:76]:
```suggestion
&prompt.cephuser;rados --pool nfs-ganesha --namespace foo put debugconf-nfs.foo debugconf-nfs.foo​
```

fixup the flow?

File: xml/bp_troubleshooting_nfs.xml [hunk 8]
```diff
@@ -13,7 +13,131 @@
...
+</screen>
+     <para>
+       Add a reference to the parameter in your &rados; common configuration file:
```
Comment (mgfritch) [inline xml/bp_troubleshooting_nfs.xml:79]:
```suggestion
       Add a reference to the debug configuration object in the &rados; common configuration file:
```

File: xml/bp_troubleshooting_nfs.xml [hunk 9]
```diff
@@ -13,7 +13,131 @@
...
+     </para>
+<screen>&prompt.cephuser;rados --pool nfs-ganesha --namespace foo put debugconf-nfs.foo debugconf-nfs.foo​
+&prompt.cephuser;rados --pool nfs-ganesha --namespace foo put conf-nfs.foo conf-nfs.foo​
```
Comment (mgfritch) [inline xml/bp_troubleshooting_nfs.xml:82]:
```suggestion
<screen>
%url "rados://nfs-ganesha/foo/export-1" ​
%url "rados://nfs-ganesha/foo/userconf-nfs-foo" ​
%url "rados://nfs-ganesha/foo/debugconf-nfs.foo"​```

File: xml/bp_troubleshooting_nfs.xml [hunk 10]
```diff
@@ -13,7 +13,131 @@
...
+</screen>
```
Comment (mgfritch) [inline xml/bp_troubleshooting_nfs.xml:83]: add a paragraph here similar to the debugconf RADOS object to put the common config changes into a RADOS object...

e.g. `Put the updated common configuration file to the &rados; common configuration object....`

File: xml/bp_troubleshooting_nfs.xml [hunk 11]
```diff
@@ -13,7 +13,136 @@
...
+<screen>
+&prompt.cephuser;rados --pool nfs-ganesha --namespace foo put debugconf-nfs.foo debugconf-nfs.foo​
+</screen>
+     <para>
+       Add a reference to the debug configuration object in the &rados; common configuration file:
+     </para>
+<screen>
+%url "rados://nfs-ganesha/foo/export-1" ​
+%url "rados://nfs-ganesha/foo/userconf-nfs-foo" ​
+%url "rados://nfs-ganesha/foo/debugconf-nfs.foo"​```
```
Comment (bk201) [inline xml/bp_troubleshooting_nfs.xml:83]:
```suggestion
%url "rados://nfs-ganesha/foo/debugconf-nfs.foo"​
```

**Review by bk201 — APPROVED** [review]

**Comment by asettle** [comment]
As per discussion with @bk201, merging so we can have online at susedoc and more reviews than just dev. This can be iterated on easily if anything changes :)

**Pull request merged**